### PR TITLE
Move all sub-chart dependencies up to riff chart

### LINKED ISSUE #69: The riff chart doesn't install kpack with Helm v2.15.x
Tried with v2.15.0 and v2.15.1, seems to just skip the kpack installation so the riff-build-controller fails since the kpack CRDs aren't available.

### PR DESCRIPTION
**Pull request by trisberg**
- this seems necessary for helm install to work with Helm 2.15

- looks like dependency resolution has changed and doesn't recurse into dependencies

- added tags for: core-runtime, knative-runtime, streaming-runtime to be able to manage multiple dependent charts

- install example for knative-runtime is:
  "helm install projectriff/riff --name riff --set tags.knative-runtime=true"

- if you need to exclude the knative chart when knative is already installed use:
  "helm install projectriff/riff --name riff --set tags.knative-runtime=true --set knative.enabled=false"

- we can revisit this when/if we move to Helm 3

Fixes #69

### CODE REVIEW

**Comment by trisberg** [comment]
I think the 2.14 behavior was a fluke. AFAIK there is no reliable support in Helm for recursing into dependent charts and resolving their dependencies. I've only found a few open issues that suggest adding this feature, like https://github.com/helm/helm/issues/2759

File: charts/riff/values.yaml [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
-keda:
-  enabled: true
-knative:
-  enabled: true
+## Uncomment the following to disable the chart even if the tag is specified
+# keda:
+#   enabled: false
+## Uncomment the following to disable the chart even if the tag is specified
+# knative:
+#   enabled: false
```
Comment (scothis) [inline charts/riff/values.yaml]: this file isn't user editable, so there really isn't a condition in which uncommenting makes sense

File: charts/riff/values.yaml [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
-      enabled: false
+## Uncomment any runtime to install the chart even if the corresponding tag isn't specified
+#  runtimes:
+#    core:
+#      enabled: true
+#    knative:
+#      enabled: true
+#    streaming:
+#      enabled: true
+tags:
```
Comment (scothis) [inline charts/riff/values.yaml]: let's also move all of the build related components under a `build` tag that is enabled by default

File: charts/riff/requirements.yaml [hunk 3]
```diff
@@ -1,11 +1,29 @@
 dependencies:
   - name: cert-manager
     condition: cert-manager.enabled
```
Comment (scothis) [inline charts/riff/requirements.yaml]: cert-manager should be installed if any of the runtime/build tags are defined

File: charts/riff/values.yaml (hunk 1 above)
Comment (trisberg) [reply charts/riff/values.yaml]: it is of you do a `helm fetch` of the chart before installing, we could change the wording to include using `--set`

File: charts/riff/values.yaml (hunk 2 above)
Comment (trisberg) [reply charts/riff/values.yaml]: ok, sounds good

File: charts/riff/requirements.yaml (hunk 3 above)
Comment (trisberg) [reply charts/riff/requirements.yaml]: hmm, maybe we could add all tags then for cert-manager dependency

Comment (scothis) [reply charts/riff/requirements.yaml]: exactly

**Pull request merged**